metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -994,6 +1014,7 @@ class MetalsLanguageServer(
     val path = params.getTextDocument.getUri.toAbsolutePath
     buffers.remove(path)
     compilers.didClose(path)
+    trees.didClose(path)
```

Comment:
So I honestly don't know, that's why I'm asking, but will there be a penalty paid at all if a user closes a file, and then re-opens it? In that case, we removed the tree from `trees` in `Trees.scala`. It's probably not a huge deal, but am more just curious if it's necessary.

Reply:
the penalty should rather be small, parsing is fast, so this is just needed to keep memory issues in check.